weblate/api/views.py

```diff
@@ -750,6 +753,143 @@ def destroy(self, request, *args, **kwargs):
         return Response(status=HTTP_204_NO_CONTENT)
 
 
+class GlossaryViewSet(WeblateViewSet, UpdateModelMixin, DestroyModelMixin):
+    """Translation projects API."""
+
+    queryset = Glossary.objects.none()
+    serializer_class = GlossarySerializer
+    lookup_field = "id"
+
+    def get_queryset(self):
+        return (
+            Glossary.objects.filter_access(self.request.user)
+            .prefetch_related("source_language")
+            .order_by("id")
+        )
+
+    def perm_check(self, request):
+        if not request.user.has_perm("project.edit"):
+            self.permission_denied(request, message="Can not manage glossary")
+
+    def term_perm_check(self, request, permission):
+        obj = self.get_object()
+        project = obj.project
+        additional_projects = obj.links.all()
+        if not request.user.has_perm(permission, project) and not any(
+            request.user.has_perm(permission, proj) for proj in additional_projects
+        ):
+            self.permission_denied(request, message="Can not manage glossary terms")
+
+    def update(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().update(request, *args, **kwargs)
+
+    def destroy(self, request, *args, **kwargs):
+        self.perm_check(request)
+        return super().destroy(request, *args, **kwargs)
+
+    @action(detail=True, methods=["get", "post"])
+    def projects(self, request, **kwargs):
+        obj = self.get_object()
+        if request.method == "POST":
+            self.perm_check(request)
+            if "project_slug" not in request.data:
+                raise ParseError("Missing 'project_slug' parameter")
+
+            project_slug = request.data["project_slug"]
+
+            try:
+                project = Project.objects.get(slug=project_slug)
+            except Project.DoesNotExist:
+                raise Http404("No project slug '%s' found!" % project_slug)
+
+            obj.links.add(project)
+            serializer = self.serializer_class(obj, context={"request": request})
+
+            return Response(data={"data": serializer.data}, status=HTTP_201_CREATED)
+
+        queryset = obj.links.order_by("id")
+        page = self.paginate_queryset(queryset)
+
+        serializer = ProjectSerializer(page, many=True, context={"request": request})
+
+        return self.get_paginated_response(serializer.data)
+
+    @action(
+        detail=True, methods=["delete"], url_path="projects/(?P<project_slug>[^/.]+)"
+    )
+    def delete_projects(self, request, id, project_slug):
+        obj = self.get_object()
+        self.perm_check(request)
+
+        try:
+            project = Project.objects.get(slug=project_slug)
+        except (Project.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=HTTP_400_BAD_REQUEST,
+            )
```

Comment:
This seems better approach:

```suggestion
            raise ParseError(str(error))
```